src/ResourceManagement/EventHub/EventHubAuthorizationRuleImpl.cs

```diff
@@ -0,0 +1,139 @@
+        {
+            this.ancestor =  new TwoAncestor(inner.Id);
+        }
+
+        ///GENMHASH:CE4FC6FAABF2BD5DC85320962AA0B51B:F74F33DC73C74BC484CB104F3A7EEC28
+        internal EventHubAuthorizationRuleImpl(string name, IEventHubManager manager) : base(name, new AuthorizationRuleInner(), manager)
+        {
+        }
+
+        ///GENMHASH:BC796D0A7EFEEA08ADC9653640ACCBCF:E2D0E9C2E913034FE2F5B8204F5797F5
+        public string NamespaceResourceGroupName()
+        {
+            return this.Ancestor().ResourceGroupName;
+        }
+
+        ///GENMHASH:D3F702AA57575010CE18E03437B986D8:1FD21C33BB571CE08A02624A543D9C3C
+        public string NamespaceName()
+        {
+            return this.Ancestor().Ancestor2Name;
+        }
+
+        ///GENMHASH:5BE993289973437601F36360EECCC6F5:44D93BEB0AEB0D232B2497DD293956E3
+        public string EventHubName()
+        {
+            return this.Ancestor().Ancestor1Name;
+        }
+
+        ///GENMHASH:0D1A56A2D4975A5117372A4C6EDA24AD:0D55D5CB70DDB4764B2683E09E4F714C
+        public EventHubAuthorizationRuleImpl WithExistingEventHubId(string eventHubResourceId)
+        {
+            this.ancestor = new TwoAncestor(SelfId(eventHubResourceId));
+            return this;
+        }
+
+        ///GENMHASH:8E558916BC9FD09AB1EE2BAA127372EB:1D4B99B2219FD6B197BBB7366BE9C73B
+        public EventHubAuthorizationRuleImpl WithExistingEventHub(string resourceGroupName, string namespaceName, string eventHubName)
+        {
+            this.ancestor = new TwoAncestor(resourceGroupName, eventHubName, namespaceName);
+            return this;
+        }
+
+        ///GENMHASH:F99F3258B7041A2910BA994BB518EAA3:2CC5F403817B58A1E7310EB6F1AF1764
+        public EventHubAuthorizationRuleImpl WithExistingEventHub(IEventHub eventHub)
+        {
+            this.ancestor = new TwoAncestor(SelfId(eventHub.Id));
+            return this;
+        }
+
+        ///GENMHASH:323E13EA523CC5C9992A3C5081E83085:B21407B848292AD05D2EDA15DA7A077A
+        protected override Task<AccessKeysInner> GetKeysInnerAsync(CancellationToken cancellationToken = default(CancellationToken))
+        {
+            return this.Manager.Inner.EventHubs
+                .ListKeysAsync(this.Ancestor().ResourceGroupName,
+                    this.Ancestor().Ancestor2Name,
+                    this.Ancestor().Ancestor1Name,
+                    this.Name,
+                    cancellationToken);
+        }
+
+        ///GENMHASH:2A78999F239DA090C8DF19A6D1F08331:C4C36D1AABF35564B2294D36E9FB136D
+        protected override Task<AccessKeysInner> RegenerateKeysInnerAsync(KeyType keyType, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            RegenerateAccessKeyParametersInner regenKeyInner = new RegenerateAccessKeyParametersInner
+            {
+                KeyType = keyType
+            };
+            return this.Manager.Inner.EventHubs
```

Comment:
async...awail

Reply:
same comment as [this](https://github.com/Azure/azure-libraries-for-net/pull/179#discussion_r170814652)